sdk/core/System.ClientModel/src/Options/LoggingOptions.cs

```diff
@@ -0,0 +1,177 @@
+
+    private const int DefaultLoggedContentSizeLimit = 4 * 1024;
+    private const bool DefaultIsLoggingEnabled = true;
+    private const bool DefaultIsLoggingContentEnabled = false;
+    private static readonly IList<string> s_defaultLoggedHeaderNames =
+        new[] {
+            "traceparent",
+            "Accept",
+            "Cache-Control",
+            "Connection",
+            "Content-Length",
+            "Content-Type",
+            "Date",
+            "ETag",
+            "Expires",
+            "If-Match",
+            "If-Modified-Since",
+            "If-None-Match",
+            "If-Unmodified-Since",
+            "Last-Modified",
+            "Pragma",
+            "Request-Id",
+            "Retry-After",
+            "Server",
+            "Transfer-Encoding",
+            "User-Agent",
+            "WWW-Authenticate" };
+    private static readonly IList<string> s_defaultLoggedQueryParameters = new[] { "api-version" };
+
+    private bool _isLoggingEnabled = DefaultIsLoggingEnabled;
+    private int _loggedContentSizeLimit = DefaultLoggedContentSizeLimit;
+    private bool _isLoggingContentEnabled = DefaultIsLoggingContentEnabled;
+    private IList<string> _loggedHeaderNames = s_defaultLoggedHeaderNames;
+    private IList<string> _loggedQueryParameters = s_defaultLoggedQueryParameters;
+    private string? _clientAssembly;
+    private string? _requestIdHeaderName;
+
+    /// <summary>
+    /// Get or sets value indicating whether HTTP pipeline logging is enabled.
+    /// </summary>
+    public bool IsLoggingEnabled
+    {
+        get => _isLoggingEnabled;
+        set
+        {
+            AssertNotFrozen();
+
+            _isLoggingEnabled = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if request or response content should be logged.
+    /// </summary>
+    public bool IsLoggingContentEnabled
+    {
+        get => _isLoggingContentEnabled;
+        set
+        {
+            AssertNotFrozen();
+
+            _isLoggingContentEnabled = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating maximum size of content to log in bytes. Defaults to 4096.
+    /// </summary>
+    public int LoggedContentSizeLimit
+    {
+        get => _loggedContentSizeLimit;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggedContentSizeLimit = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a list of header names that are not redacted during logging.
+    /// </summary>
+    public IList<string> LoggedHeaderNames
+    {
+        get => _loggedHeaderNames;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggedHeaderNames = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a list of query parameter names that are not redacted during logging.
+    /// </summary>
+    public IList<string> LoggedQueryParameters
+    {
+        get => _loggedQueryParameters;
+        set
```

Comment:
TODO: Does it make more sense for there to not be setters on the IList<string> parameters?

Reply:
That would be consistent with FDG guidelines for collection properties, I think.